### DISCUSSION TITLE
반응형 웹 디자인과 적응형 웹 디자인

### DISCUSSION THREAD

**Discussion by WooGie911** [wanted-9th-3team/frontend-study#84] [프론트 기술 질문 공간]
반응형 웹 디자인과 적응형 웹 디자인에 대해 얼마나 알고 있나요?
반응형 웹 디자인과 적응형 웹 디자인에 대해 설명 해주세요.

**Comment by michoball** [comment]
두개의 디자인 방법 모두 예전 각 기기 사이즈에 맞게 디자인을 따로 하던 방식의 단점을 보완하고자 나왔습니다. 

기본적으로 사용자가 보는 화면크기 ( viewPort ) 에 따라 디자인이 바뀌도록 하여 동일한 UX를 제공하고자 합니다. 

반응형은 하나의 코드 베이스를 가지는 viewPort 를 기준으로 변경되는 사이즈( break points )에 따라 요소들의 크기를 조절하는 디자인입니다.  사이즈가 변경되어 요소들이 변하더라도 그에 구애받지 않는 UX를 제공합니다. 
 하나의 코드 베이스에서 viewPort 에 따른 변경만 가져가능 방식이라 스타일링 코드가 더 길어서 페이지 로드에 영향을 줄 수 있고, 모든 기기에 맞춤으로 하기 힘들기 때문에 특정 기기에서는 UX가 좋지 못할 수 있습니다. 

적응형은 화면 사이즈에 따라 디자인한 정적인 레이아웃을 제공하는 방식입니다. 
user가 해당 사이트에 접속하면 user가 사용하는 기기의 사이즈를 인식하여 해당 사이즈에 맞는 ux design을 제공하는 방식입니다. 
 각 사이즈별 기기중 사용자가 가장 빈번히 접근하는 사이즈 ( 예를 들어 모바일 이나 테블릿으로 많이 해당 사이트에 들어온다면 -> 320px, 760px 이 중요한 사이즈가 되는 식 )의 UX를 더 신경써서  디자인하고는 합니다.  그래서 특정 기기에서 웹 사이트의 UX가 다를 수 있습니다.  또한 디자이너가 정한 몇 개의 viewPort에 만 해당하는 디자인을 제공하기 때문에 이에 맞지않는 viewPort에서는 UX가 좋지 못할 수 있습니다.

**Comment by WooGie911** [comment]
반응형 웹 디자인은 감지된 화면 크기에 따라 자동으로 페이지가 재배열 되는 유동적인 접근 방식입니다. 반면, 적응형 웹 디자인은 브라우저가 주어진 플랫폼에 맞춰 틀별히 생성된 레이아웃을 불러오는 웹 디자인 유형입니다.

반응형 웹 디자인을 사용하면 웹 페이지의 디자인과 레이아웃이 모든 화면 크기에 자동으로 맞춰집니다. 
보통 CSS 미디어 쿼리를 사용하여 사용자가 사용하는 기기의 특성을 검사하고 그 내용을 바탕으로 자체적으로 렌더링됩니다.
모든 플랫폼에서 일관된 컨텐츠를 경험할수 있으며, 보편적이지 않은 화면 크기의 기기에서도 똑같이 작동한다는 장점이 있습니다.
다만 각 기기에서 렌더링 되는 방식에 대한 통제가 상대적으로 어려우며, 디자인에 대한 더 많은 전문 지식이 필요하여 플랫폼 간의 많은 테스트및 디자인 조정이 필요합니다. 결과적으로 동적 컨텐츠를 불러오는데 더 많은 작업이 요구되어 웹사이트의 성능이 저하까지 가져올수 있습니다.

적응형 웹 디자인은 웹 페이지에서 감지된 기기를 기반으로 미리 만들어진 정적인 레이아웃을 불러옵니다. 이를 위해선 디자이너가 화면 너비에 맞춰 별도의 디자인 작업을 해야하며 일반적으로 320px,  480px,  760px,  960px,  1200px,  1600px 의 너비를 사용합니다.
장점으로는 각 플랫폼 및 상황과 맥락에 맞춰 제공되기 때문에 사용자 입장에서 불편함을 느끼지 않고, 원하는 기기에 최적화된 디자인을 할수있는 고성능 작업을 수행할수 있으며, 광고 및 타사와의 통합 컨텐츠 등 맞춤화에 유리합니다.
다만 컨텐츠가 모든 기기에서 일관되지 않을 경우 SEO에 부정적인 영향을 미칠수 있습니다.

간단하게 정리하면 반응형 웹 디자인은 사용되는 모든 컨텐츠를 다운받은 뒤 현재 해상도에 맞는 화면에 랜더링되며 PC든 Mobile이든 모든 리소스를 다운 받기에 로딩속도가 느립니다.
반면 적응형 웹 디자인은 감지된 디바이스에 필요한 컨텐츠만 다운받은 뒤 화면에 렌더링되기 떄문에 PC로 접속하면 Mobile리소스는 다운받지 않고 PC리소스만 받기 때문에 로딩속도가 빠릅니다.
